types/index.d.ts

```diff
@@ -0,0 +1,109 @@
+// Minimum TypeScript Version: 3.8
+/// <reference types="node" />
+import type { ConnectionOptions } from 'typeorm';
+import type { IncomingMessage, ServerResponse } from 'http';
+import type { PossibleProviders } from './providers';
+import Adapter from './adapters';
+
+interface InitOptions {
+  site: string;
+  providers: PossibleProviders[];
```

Comment:
Each of these providers are typed to return `void` when actually they return an object ([email provider implementation](https://github.com/nextauthjs/next-auth/blob/main/src/providers/email.js#L5))

In the current state, the following example from the [docs](https://next-auth.js.org/getting-started/example#add-api-route) generates the following error

```typescript
import type { NextApiRequest, NextApiResponse } from "next";
import NextAuth from "next-auth";
import Providers from "next-auth/providers";

const options = {
	providers: [
		Providers.GitHub({
			clientId: process.env.GITHUB_ID!,
			clientSecret: process.env.GITHUB_SECRET!,
		}),
	],
};

export default (req: NextApiRequest, res: NextApiResponse) => NextAuth(req, res, options)

```

```
Error:(14, 82) TS2345: Argument of type '{ providers: void[]; }' is not assignable to parameter of type 'InitOptions'.
  Types of property 'providers' are incompatible.
    Type 'void[]' is not assignable to type 'PossibleProviders[]'.
      Type 'void' is not assignable to type 'PossibleProviders'.
```

Reply:
This was already updated in the [actual types PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/45775/files) 👍🏽